server/src/main/java/org/apache/druid/server/coordinator/duty/BalanceSegments.java

```diff
@@ -144,16 +144,21 @@ private void balanceTier(
     }
 
     final int maxSegmentsToMove = Math.min(params.getCoordinatorDynamicConfig().getMaxSegmentsToMove(), numSegments);
-    int decommissioningMaxPercentOfMaxSegmentsToMove =
-        params.getCoordinatorDynamicConfig().getDecommissioningMaxPercentOfMaxSegmentsToMove();
-    int maxSegmentsToMoveFromDecommissioningNodes =
-        (int) Math.ceil(maxSegmentsToMove * (decommissioningMaxPercentOfMaxSegmentsToMove / 100.0));
-    log.info(
-        "Processing %d segments for moving from decommissioning servers",
-        maxSegmentsToMoveFromDecommissioningNodes
-    );
-    Pair<Integer, Integer> decommissioningResult =
-        balanceServers(params, decommissioningServers, activeServers, maxSegmentsToMoveFromDecommissioningNodes);
+
+    // Only do the work of calling balanceServers to move from decomissioning servers if there are > 0 decomissioning servers
+    Pair<Integer, Integer> decommissioningResult = new Pair<>(0, 0);
+    if (!decommissioningServers.isEmpty()) {
```

Comment:
Would it make sense to add this check inside `balanceServers()` method itself?
It already has a check as follows:
```
if (maxSegmentsToMove <= 0) {
      return new Pair<>(0, 0);
    }
```

We could add a new else clause `else if (toMoveFrom.isEmpty() || toMoveTo.isEmpty())` to the above `if`.

Reply:
hmm, I guess I am indifferent. But I'd also pose the question as to why balanceServers is even called here when segments to move was calculated to be 0 before the call. I guess it is kind of a programmers preferred style. Leaving the argument checks to the method being called instead of the code calling. I think I'm fine with changing it to match existing style. I will try to get it done sometime today. Thanks for taking a look!